docs/examples/nft-marketplace/nft-application.md

```diff
@@ -0,0 +1,710 @@
+        return state;
+    }
+    ```
+
+4. Now let's start writing the `Home` page:
+
+    ```typescript
+
+    import { Loader } from 'components';
+    import { useNFTs } from 'hooks/api';
+    import styles from './Home.module.scss'
+
+    function Home() {
+
+    const nfts = useNFTs();
+    const isAnyNft = !!nfts?.length;
+
+    return (
+        <>
+        <header className={styles.header}>
+            <h2 className={styles.heading}>NFTs</h2>
+        </header>
+        {nfts ? (
+            <>
+            {isAnyNft && <ul className={styles.list}>Display NFTs here</ul>}
+            {!isAnyNft && <h2>There are no NFTs at the moment</h2>}
+            </>
+        ) : (
+            <Loader />
+        )}
+        </>
+    )
+    }
+    export { Home };
+    ```
+
+5. We read `nfts` using the previously written hook `useNFTs`:
+
+    ```typescript
+    const nfts = useNFTs();
+    ```
+
+6. Then we check whether the contract has tokens:
+
+    ```typescript
+    const isAnyNft = !!nfts?.length;
+    ```
+
+7. Let's create a component that will display NFT:
+
+    ```
+    mkdir pages/home/nft
+    touch pages/home/nft/nft.tsx
+    ```
+
+    and write the component:
+
+    ```typescript
+    import { Link } from "react-router-dom";
+    import { getIpfsAddress } from "utils";
+    import styles from './nft.module.scss'
+
+    type Props = {
+        id: string;
+        name: string;
+        media: string
+    }
+
+    function NFT( {id, name, media }: Props) {
+        const to = `/nft/${id}`;
+        const src = getIpfsAddress(media)
+        const text = `#${id}`
+        return (
+            <Link to={to} className={styles.nft}>
+                <img src={src} alt={name}/>
+                <h3 className={styles.heading}>{name}</h3>
+                <p className={styles.text}>{text}</p>
+            </Link>
+        )
+    }
+
+    export { NFT };
+    ```
+
+8. Then let's write a function for retrieving all NFTs from the contract in the `Home.tsx` file:
+
+    ```typescript 
+    ...
+    import { NFT } from './nft/nft';
+
+    function Home() {
+
+    const nfts = useNFTs();
+    const isAnyNft = nfts?.length;
+
+    const getNFTs = () => 
+        nfts?.map( ({name, id, media}) => (
+        <li key={id}>
+            <NFT id = {id} name = {name} media = {media} />
+        </li>
+        ))
+        
+    ...
+    }
+    export { Home };
+    ```
+    The whole code of the `Home` page:
+    ```typescript
+    import { Loader } from 'components';
+    import { useNFTs } from 'hooks/api';
+    import styles from './Home.module.scss'
+    import { NFT } from './nft/nft';
+
+    function Home() {
+
+    const nfts = useNFTs();
+    const isAnyNft = nfts?.length;
+
+    const getNFTs = () => 
+        nfts?.map( ({name, id, media}) => (
+        <li key={id}>
+            <NFT id = {id} name = {name} media = {media} />
+        </li>
+        ))
+        
+        
+    const NFTs = getNFTs();
```

Comment:
Remove this assignment. Unnecessary cuz we can just invoke function in a jsx, and confusing cuz we already have `nfts` name